build/build_atchem2.sh

```diff
@@ -1,60 +1,72 @@
-#!/bin/sh
-# -----------------------------------------------------------------------------
-#
-# Copyright (c) 2017 Sam Cox, Roberto Sommariva
-#
-# This file is part of the AtChem2 software package.
-#
-# This file is covered by the MIT license which can be found in the file
-# LICENSE.md at the top level of the AtChem2 distribution.
-#
-# -----------------------------------------------------------------------------
-
-# Script to process the mechanism file (*.fac) and convert it to the
-# correct format for AtChem2.
-#
-# N.B.: the script MUST be run from the base directory of AtChem2.
-#
-# $1 is the location of the chemical mechanism file in FACSIMILE
-#    format. This argument is NOT optional, and there is no
-#    default.
-#
-# $2 is the directory for the chemical mechanism in Fortran format:
-#    - mechanism.species
-#    - mechanism.reac
-#    - mechanism.prod
-#    - mechanism.ro2
-#    - mechanism.f90
-#    By default, it is ./model/configuration/
-#
-# $3 is the directory of the MCM data files:
-#    - list of organic peroxy radicals (RO2)
-#    - parameters to calculate photolysis rates
-#    By default, it is ./mcm/
-
-set -e
-echo ''
-echo "* facsimile mechanism file:" $1
-echo "* fortran mechanism directory [ default = ./model/configuration/ ]:" $2
-echo "* mcm data files directory [ default = ./mcm/ ]:" $3
-
-echo ''
-echo "call mech_converter.py"
-python ./build/mech_converter.py $1 $2 $3
-
-echo ''
-if [ -z $2 ]; then
-  echo "make sharedlib"
-  make sharedlib
-  echo "=> shared library directory: ./model/configuration/"
-else
-  echo "make sharedlib" $2
-  make sharedlib SHAREDLIBDIR=$2
-  echo "=> shared library directory:" $2
-fi
-
-echo ''
-echo "make base executable"
-make
-
-exit 0
+#!/bin/sh

```

Comment:
I've made this change but there still appears to be a problem. There is a syntax error on the curly bracket of line:
f = open(f"./{mech_file}","a+")

I've come across this exact error when trying to use python 2.7... is Travis compatible with python 3?

Reply:
Travis can use Python 3, but I think by default it's using 2.7 as that's what most systems will use if you just use `python` rather than `python3`.

We don't stipulate that users need python3 - it's better to maintain compatibility, hence for example the use of `from __future__ import print_function` in `mech_converter.py`. It may be useful to not use the `f" "` syntax, as that is python3-only, to maintain 2.7-compatibility. It would also match the syntax we use elsewhere in e.g. `mech_converter.py`.

Perhaps use the syntax (also using the `with` syntax which is more modern):
```
with f as open(mech_file,"a+")
    for x in spec:
        x = x.split()
        z = (x[1])
        f.write('% ' + str(y) + ' : ' + str(z) + ' = LOSS ;\n')
```